components/x11/xorg-server/files/x11-init

```diff
@@ -0,0 +1,42 @@
+#! /usr/bin/sh -e
+#
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2019 Alexander Pyhalov
+#
+
+# List of directories to create, splitted by ";"
+DIRS="/tmp/.ICE-unix:root:root:1777"
+
+
+. /lib/svc/share/smf_include.sh
+
+PATH=/bin
+
+IFS=";"
+for rec in $DIRS; do
+   dir=`echo $rec | cut -d ":" -f 1`
+   owner=`echo $rec | cut -d ":" -f 2`
+   group=`echo $rec | cut -d ":" -f 3`
+   mode=`echo $rec | cut -d ":" -f 4`
```

Comment:
ETOOMANYFORKS :)

````
$ ksh -c 'DIRS="/tmp/.ICE-unix:root:root:1777;/etc/qwe:jim:staff:555"; echo "$DIRS" | tr ";" "\n" | while IFS=":" read dir owner group mode in $rec ; do echo "$dir = $owner = $group = $mode" ; done'
````
Could also define DIRS as a multiline string right away and so no `tr`, and even can do no `echo`:

````
DIRS="/tmp/.ICE-unix:root:root:1777
/tmp/.ICE-unix:root:root:1777"

while IFS=":" read dir owner group mode in $rec ; do
    echo "$dir = $owner = $group = $mode"
done << EOF 
$DIRS
EOF
````